CRM/CiviAwards/Service/AwardImport.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+use CRM_CiviAwards_Helper_CaseTypeCategory as CaseTypeCategory;
+
+/**
+ * Utility class for importing Awards using CSV importer extension.
+ */
+class CRM_CiviAwards_Service_AwardImport {
+
+  /**
+   * Create an Award and the related entities.
+   *
+   * @param array $params
+   *   Information for creating the award and related entities.
+   */
+  public function create(array $params) {
+    $tx = new CRM_Core_Transaction();
+
+    $params['case_type_category'] = $this->getCaseCategoryValueForAwards();
+    $params['definition'] = $this->createDefinition($params);
+    try {
+      $caseType = civicrm_api3('CaseType', 'create', $params);
+    }
+    catch (Exception $exception) {
+      $tx->rollback();
+
+      throw new API_Exception('Exception while saving the Case Type for Award: ' . $exception->getMessage());
+    }
+    $caseTypeId = array_shift($caseType['values'])['id'];
+
+    $params['case_type_id'] = $caseTypeId;
+    $params['award_manager'] = !empty($params['award_manager']) ? json_decode($params['award_manager'], TRUE) : [];
+    $params['review_fields'] = !empty($params['review_fields']) ? json_decode($params['review_fields'], TRUE) : [];
+    try {
+      civicrm_api3('AwardDetail', 'create', $params);
+    }
+    catch (Exception $exception) {
+      $tx->rollback();
+
+      throw new API_Exception('Exception while saving the AwardDetail: ' . $exception->getMessage());
+    }
+
+    if (!empty($params['review_panel_title'])) {
```

Comment:
We can take out the review panel creation part like I mentioned in other comments. The idea is to be able to import an Award with basic details to get it working and other complex parts can be filled in later via the UI.